.circleci/config.yml

```diff
@@ -0,0 +1,120 @@
+version: 2.1
+orbs:
+  aws-cli: circleci/aws-cli@1.3.1
+
+commands:
+  check-for-changes:
+    description: "Searches for any files matching the pattern in a diff result"
+    parameters:
+      in:
+        description: "Returns true if this pattern matches any folder/file in the diff"
+        type: string
+      since:
+        description: "The git commit that we are checking against"
+        type: string
+    steps:
+      - run:
+          command: |
+            git diff --name-only HEAD << parameters.since >> | grep << parameters.in >>
+
+  write-build-info:
+    description: "Write relevant info to the current build/deploy to a file for referencing"
+    parameters:
+      data:
+        description: "The contents of the build file"
+        type: string
+        default: |
+          BUILD_COMMIT=<< pipeline.git.revision >>
+      dest:
+        description: "The destination of the build file"
+        type: string
+    steps:
+      - run:
+          name: "Writing build info"
+          command: |
+            echo "<< parameters.data >>" > << parameters.dest >>/BUILD.txt
+
+  copy-to-s3:
+    description: "Copies adirectory to a bucket (and folder) in S3"
+    parameters:
+      source:
+        description: "The local directory to copy"
+        type: string
+      bucket:
+        description: "The S3 bucket (and optional folder) to copy everything to"
+        type: string
+    steps:
+      - run: aws s3 cp << parameters.source >> s3://<< parameters.bucket >> --recursive --acl public-read
+
+jobs:
+  needs-to-publish-storybook:
+    docker:
+      - image: circleci/node:14.5.0
+    parameters:
+      doc_site:
+        type: string
+        description: "The domain for the doc site containing the build file"
+    steps:
+      - checkout
+      - run:
+          name: "Get last published commit"
+          command: |
+            echo "export LAST_PUBLISHED_COMMIT=$(curl -s -f https://<< parameters.doc_site >>/storybook/BUILD.txt | sed -r 's/^.*=(.*)$/\1/')" >> $BASH_ENV
+            source $BASH_ENV
+        # If we didn't find a last published commit, then we need to publish
+      - run: |
+          if [ "$LAST_PUBLISHED_COMMIT" = "" ]; then
+            circleci-agent step halt
+          fi
+      - check-for-changes:
+          in: "formation-react"
+          since: $LAST_PUBLISHED_COMMIT
```

Comment:
This makes sense to me now, but can we explain this directly in the config file so our future selves—and others—will know what the reasoning is later?

I feel like this would have saved us a lot of headache for the Metalsmith build, so I'd like to learn from that and document directly in the code here. I find commenting more on the "why" is typically more helpful than the "what," but given that this is relatively esoteric (from an FE's perspective, that is), it's probably useful to leave comments on what this is doing in here too.